packages/ref-core/src/cmip_ref_core/pycmec/metric.py

```diff
@@ -167,10 +178,18 @@ def _check_nested_dict_keys(cls, nested: dict[Any, Any], metdims: dict[Any, Any]
             elif isinstance(value, dict) and level < len(metdims[MetricCV.JSON_STRUCTURE.value]) - 1:
                 cls._check_nested_dict_keys(value, metdims, level + 1)
             elif isinstance(value, dict):
-                tmp = dict(value)
-                if MetricCV.ATTRIBUTES.value in tmp:
-                    tmp.pop(MetricCV.ATTRIBUTES.value)
-                StrNumDict(tmp)
+                raise ValueError(
+                    f"Reached the last value {key} of the json_structure,\n"
+                    "but find there are some deeper dictionaries under it"
+                )
+            elif (not isinstance(value, float)) and (not isinstance(value, int)):
+                raise ValueError("Metric values have to be numeric types (float or int)")
```

Comment:
FYI this is different to the type declared by `StrNumDict` (and preferable for the frontend because it is easier to format). Is that intentional?

i.e all leaf values must be floats or int

Reply:
Yes. We want to make sure the values in the deepest dictionary (i.e., the calculated metrics) are numeric values (i.e. float or int). Previously, we may not enforce the rule.